torchtune/data/_collate.py

```diff
@@ -214,6 +214,142 @@ def padded_collate_sft(
     return {"tokens": input_ids.long(), "labels": labels.long()}
 
 
+def padded_collate_tiled_images_with_cross_attention(
+    batch: List[Dict[str, Any]],
+    padding_idx: int = 0,
+    ignore_idx: int = CROSS_ENTROPY_IGNORE_IDX,
+) -> Dict[str, torch.Tensor]:
+    """Pad a batch of text sequences, tiled image tensors, aspect ratios,
+    and cross attention masks.
+
+    ``batch`` is expected to be a list of sample dicts containing the following:
+        - "tokens": List[int] of length text_seq_len, varies across samples
+        - "labels": List[int] of length text_seq_len, varies across samples
+        - "images": List[Tensor], each with shape (n_tiles, c, h, w)
+        - "encoder_mask": List[Tensor], each with shape (text_seq_len, image_seq_len)
+        - "aspect_ratio": List[Tensor], each with shape (h_ratio, w_ratio)
```

Comment:
The shape is just (2,) right? Also maybe worth saying that len(images) == len(encoder_mask) == len(aspect_ratio) for each element in the batch